### PR TITLE
refactor(server): replace node-resque with kue

### LINKED ISSUE #363: Replace node-resque with kue
Just need something maintained that works, and kue does have much better docs.

### PR DESCRIPTION
**Pull request by 6a68**
Fixes #363.

### CODE REVIEW

File: server/work-queue/jobs/create-visit.js [hunk 1]
```diff
@@ -7,16 +7,15 @@
...
-  perform: function(o, cb) {
-    log.verbose('job created with params ' + JSON.stringify(o));
-    visit.create(o.userId, o.visitId, o.visitedAt, o.url, o.urlHash, o.title, function (err) {
-      log.verbose('inside the visit.create callback inside the createVisit job!');
-      if (err) {
-        log.warn('failed at visit.create step for visit url ' + o.url + ' :' + err);
-      }
-      cb(err);
+  work: function(queue) {
+    queue.process('createVisit', 10, function(job, done) {
```
Comment (6a68) [inline server/work-queue/jobs/create-visit.js]: Would be cool to put this in config at some point

File: server/work-queue/jobs/extract-page.js [hunk 2]
```diff
@@ -7,45 +7,48 @@
...
-          cb(err);
+          log.verbose('succeeded at embedly.extract step', d.url);
+          // the visit creation job has probably created a record in the user_page table.
+          // if not, update will lazily create it.
+          userPage.update(d.userId, d.url, d.urlHash, d.title, extracted, function (err) {
+            var msg;
+            if (err) {
+              msg = 'failed at userPage.update step';
+              log.warn(msg, d.url);
+              done(new Error(msg));
```
Comment (pdehaan) [inline server/work-queue/jobs/extract-page.js]: Does this need to be `return done()` here (otherwise `done()` could be called twice on err.

Comment (6a68) [inline server/work-queue/jobs/extract-page.js]: dang, yup.

**Pull request merged**